tests/integration-test/server/index.js

```diff
@@ -0,0 +1,258 @@
+        `file must be set via TEST_INPUT_FILE environment variable.`);
+    }
+
+    TEST_DOCKER_IMAGE = process.env.TEST_DOCKER_IMAGE;
+    if (!TEST_DOCKER_IMAGE) {
+      throw new Error(`A valid, local docker image must be set via ` +
+        `TEST_DOCKER_IMAGE environment variable.`);
+    }
+
+    GIVEN_ANALYTIC_FILE = process.env.TEST_ANALYTIC_FILE;
+    if (!GIVEN_ANALYTIC_FILE) {
+      throw new Error(`A valid absolute path to the desired analytic JSON ` +
+        `test file must be set via TEST_ANALYTIC_FILE environment variable.`);
+    }
+
+    PORT = 4000;
+    debug('Environment variable validation complete.');
+
+    STORAGE_BASE_DIR = path.join(__dirname, './storage');
+    API_TOKEN = 'my-very-special-key12345';
+    JOB_ID = uuid4();
+    const analyticFields = await parseAndVerifyAnalyticJSON();
+    // TODO add support for parameters?
+    const {taskURL, task} = await generateTaskJSON(analyticFields);
+    debug('Test setup complete.');
+
+    debug('Spinning up mock server.');
+    const socket = await server.spinup(task);
+
+    const dockerCmd = await generateDockerCommand(taskURL);
+    console.log('\n\nRun the following docker command to test your image now:\n\n',
+      dockerCmd, '\n\nCleanup generated files via npm run clean or ' +
+      'yarn run clean.\n\n');
+
+    process.on('SIGTERM', shutdown(socket));
+    process.on('SIGINT', shutdown(socket));
+    process.on('unhandledRejection', (reason, p) => {
+      console.error('Unhandled rejection at Promise: ', p, 'reason:', reason);
+      console.error(reason.stack);
+    });
+  } catch (err) {
+    console.error(err);
+    process.exit(1);
+  }
+
+  function generateDockerCommand(taskURL) {
+    return new Promise(function(resolve, reject) {
+      debug('Generating the docker run command.');
+      var cmd = 'docker run ';
+      if (process.env.TEST_USE_GPU) {
+        cmd += '--runtime=nvidia ';
+      }
+      cmd += `--name ${JOB_ID} ` +
+        `-e TASK_DESCRIPTION="${taskURL}" ` +
+        `-e ENV="LOCAL" ` +
+        `-e API_TOKEN="${API_TOKEN}" ` +
+        `-e OS="${process.platform}" ` +
+        `--network="host" ${TEST_DOCKER_IMAGE}; ` +
+        `echo -e "\n\nNow close the server (Ctrl-C) to finish the ` +
+        `test and retrive your results!"`;
+      debug('Docker command generation complete.');
+      return resolve(cmd);
+    });
+  }
+
+  async function generateTaskJSON(analyticFields, params={}) {
+    debug('Generating task JSON file using analytic fields:', analyticFields);
+    const inputFilename = path.basename(GIVEN_INPUT_FILE);
+    const inputFilepath = path.join(STORAGE_BASE_DIR, inputFilename);
+    debug(`Copying analytic file to new location, ${inputFilepath}.`);
+    await pExec(`cp ${GIVEN_INPUT_FILE} ${inputFilepath}`);
+    const task = {
+      analytic: analyticFields.name,
+      version: analyticFields.version,
+      job_id: JOB_ID,
+      inputs: {
```

Comment:
Suggestions:
(1) the user provides a JSON file like:
```
{
    "inputs": {
        "<input1>": "<path1>",
        ...
    },
    "parameters": {
        "<param1>": <val1>,
        ...
    }
}
```
and runs like this:
```
bash run_test.bash --config '<path>' ...
```
(2) user sets inputs/parameters via commandline flags:
```
bash run_test.bash \
    --input '<input1>=<path1>' --input '<input2>=<path2>' ... \
    --parameter '<param1>=<val1>' ...
   ...
```
Even better (and not hard)- support both. let the user optionally load a config and overwrite values via commandline args if they want to.

Reply:
We need to implement this. I am fine with just doing option (2) for now. It should be straightforward. If there is a problem with repeated command-line flags, then we could use `--inputs '<input1>=<path1>,<input2>=<path2>,...` to support multiple inputs.

This is important because user's analytics can have any number of analytics, and the input/output names are arbitrary. They are also free to define parameters for their analytics.

An important component of this is validating that the user provided all required inputs and parameters, as defined in their analytic JSON file.